### PR TITLE
feat: format commented parenthesized expr/patterns

### PR DESCRIPTION
**Pull request by QuadnucYard**
This PR includes:
- feature: parenthesized expressions can be formatted just like arrays (the rules are similar to array). Nested parentheses will not be removed if there exists any comment as a direct child.
- feature: remove unnecessary parentheses if the inner expression is literal, array, dict, destructuring, block, or pattern. For safety, parens around idents are kept.
- change: destructuring and params with comments are no longer forced to fold into one line.
- feature: format closures with comments, which was left out in the previous PR.

### CODE REVIEW

File: tests/snapshots/assets__check_file@tablex.typ-120.snap [hunk 1]
```diff
@@ -743,13 +749,19 @@ snapshot_kind: text
 
 // Returns 'true' if the cell at (x, y)
 // exists in the grid.
-#let grid-has-pos(grid, x, y) = (grid-index-at(x, y, grid: grid) < grid.items.len())
+#let grid-has-pos(grid, x, y) = (
+  grid-index-at(x, y, grid: grid) < grid.items.len()
```
Comment (QuadnucYard) [inline tests/snapshots/assets__check_file@tablex.typ-120.snap:753]: This respects the multiline flavor

File: tests/snapshots/assets__check_file@tablex.typ-120.snap [hunk 2]
```diff
@@ -1919,11 +1933,13 @@ snapshot_kind: text
...
-                or v.end >= cell.y + cell.rowspan  // ends at or after this cell
-            )
+          v.end in (auto, none) // always goes towards the bottom
+          or v.end >= cell.y + cell.rowspan // ends at or after this cell
+        )
 
-        (at_left_or_right and vline_hasnt_already_ended)
+        (
+          at_left_or_right and vline_hasnt_already_ended
+        )
```
Comment (QuadnucYard) [inline tests/snapshots/assets__check_file@tablex.typ-120.snap:1942]: There is a line break before `and`:
```
            (at_left_or_right
                and vline_hasnt_already_ended)
```

**Comment by Enter-tainer** [comment]
> feature: parenthesized expressions can be formatted just like arrays

Could you elaborate?

**Review by Enter-tainer — APPROVED** [review]
I went through all the snapshot changes and I didn't see anything super weird. So, LGTM

**Comment by QuadnucYard** [comment]
> > feature: parenthesized expressions can be formatted just like arrays
> 
> Could you elaborate?

A parenthesized expression differs from an array only in that it only has one item, the delimiter is optional if single line, and no trailing comma.

**Pull request merged**